spec.bs

```diff
@@ -810,6 +810,68 @@ To <dfn>compute protocol matches a special scheme flag</dfn> given a [=construct
   1. If the result of running [=protocol component matches a special scheme=] given |protocol component| is true, then set |parser|'s [=constructor string parser/protocol matches a special scheme flag=] to true.
 </div>
 
+<h2 id=urlpatternlist-class>The {{URLPatternList}} class </h2>
+
+<xmp class="idl">
+[Exposed=(Window,Worker)]
+interface URLPatternList {
+  constructor(sequence<URLPattern> patterns);
+
+  boolean test(optional URLPatternInput input = {}, optional USVString baseURL);
+
+  URLPatternListResult? exec(optional URLPatternInput input = {}, optional USVString baseURL);
+};
+
+dictionary URLPatternListResult : URLPatternResult {
+  URLPattern pattern;
+};
+</xmp>
+
+Note: The {{URLPatternList}} is a utility that combines many patterns into one
+matcher. This functionality can also be achieved in a naive userland
+implementation that iterates over a list of patterns and matches each one
+individually. This has the unfortunate effect of not being very fast. With the
+built-in {{URLPatternList}} implementers can use an optimized implementation for
+matching under the hood which can drastically improve performance over the naive
+userland implementation.
+
+Each {{URLPatternList}} has an associated <dfn for=URLPatternList>pattern list</dfn>, a [=list=] of {{URLPattern}}s.
+
+<div algorithm>
+  The <dfn constructor for=URLPatternList lt="URLPatternList(patterns)">new URLPatternList(|patterns|)</dfn> constructor steps are:
+
+  1. [=list/For each=] |pattern| in |patterns|, run the following steps:
```

Comment:
Doesn't this imply that the pattern list is traversed as an array and is potentially slower? https://github.com/WICG/urlpattern/issues/30#issuecomment-1089229462

Reply:
The iteration steps here and elsewhere don’t appear to be observable, so engines can potentially optimize them however they see fit, right? Spec steps usually aim for the simplest explanation of observable behaviors (because observable behaviors are the only things they _can_ actually specify, further complexity is noise).

Reply:
I see, thanks for the explanation!